### PR TITLE
Support OTEL_METRIC_EXPORT_INTERVAL and OTEL_METRIC_EXPORT_TIMEOUT

### LINKED ISSUE #3417: Support `OTEL_METRIC_EXPORT_INTERVAL` and `OTEL_METRIC_EXPORT_TIMEOUT`
# Feature Request

Support for `OTEL_METRIC_EXPORT_INTERVAL` and `OTEL_METRIC_EXPORT_TIMEOUT` environmental variables.

Reference: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#periodic-exporting-metricreader

**Is your feature request related to a problem?**

See: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/900

**Describe the solution you'd like:**

 `OTEL_METRIC_EXPORT_INTERVAL` and `OTEL_METRIC_EXPORT_TIMEOUT`  can be used to set `PeriodicExportingMetricReaderOptions` values if these were not programmatically set to non-default values.

**Describe alternatives you've considered.**

Implement it only on the https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation.

## Additional Context

It can be handy for the users and it will also be reused in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation

It should be easy to implement it here: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Internal/PeriodicExportingMetricReaderHelper.cs

### PR DESCRIPTION
**Pull request by pellared**
Fixes #3417.

## Changes

The `MetricReaderOptions` defaults can be overridden using
  `OTEL_METRIC_EXPORT_INTERVAL` and `OTEL_METRIC_EXPORT_TIMEOUT`
  environmental variables as defined in the
  [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.12.0/specification/sdk-environment-variables.md#periodic-exporting-metricreader).

This change applies to all metrics exporters that use `PeriodicExportingMetricReaderHelper.CreatePeriodicExportingMetricReader`
- OTLP
- Console
- InMemory

I think it is not possible to implement it for all instances of `PeriodicExportingMetricReader` without changing and breaking its public API (assuming that the value in "options" has precedence over the environmental variable which is the currently followed pattern).

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] ~~Design discussion issue #~~
* [ ] Changes in public API reviewed
  - I added default values to `CreatePeriodicExportingMetricReader` that match the OTel spec

### CODE REVIEW

File: src/OpenTelemetry/Internal/PeriodicExportingMetricReaderHelper.cs
Comment (RassK) [inline src/OpenTelemetry/Internal/PeriodicExportingMetricReaderHelper.cs:52]: Might be worth mentioning the priority in docs? (currently I see only 2 & 3)
1. user code
2. envs
3. defaults

Comment (pellared) [reply src/OpenTelemetry/Internal/PeriodicExportingMetricReaderHelper.cs:52]: I now realized that I need to describe the env vars in `README.md` files

Comment (pellared) [reply src/OpenTelemetry/Internal/PeriodicExportingMetricReaderHelper.cs:52]: https://github.com/open-telemetry/opentelemetry-dotnet/pull/3424/commits/bd025a4d509ab66cb6f754e1b72f21ad9d7dc6db

Comment (RassK) [reply src/OpenTelemetry/Internal/PeriodicExportingMetricReaderHelper.cs:52]: > The `Options` type setters take precedence over the environment variables

👍 seems good enough

**Comment by pellared** [comment]
**Side note:** 

These failures are awkward: https://github.com/open-telemetry/opentelemetry-dotnet/runs/7217170744?check_suite_focus=true

Did anyone encounter the same problem? Looks like some strange side-effect - I have not analyzed it.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md
Comment (joaopgrassi) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:21]: Would it still make sense to use the actual type name instead of the generic `Options`? Just thinking if this won't make things harder for people looking at these instructions. How do I know the type I can use? Need to dig into the code maybe? 🤔

Comment (joaopgrassi) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:21]: Sounds good, was more just some food for though!

**Review by joaopgrassi — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]
LGTM.

**Comment by reyang** [comment]
The environment variables are still marked as Experimental 
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md, what's the plan here (e.g. push the change to a feature branch; wait until the spec becomes stable, merge in the main branch and do not release stable until the spec becomes stable)?

**Review by reyang — CHANGES_REQUESTED** [review]
https://github.com/open-telemetry/opentelemetry-dotnet/pull/3424#issuecomment-1180639030

**Comment by cijothomas** [comment]
> The environment variables are still marked as Experimental https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md, what's the plan here (e.g. push the change to a feature branch; wait until the spec becomes stable, merge in the main branch and do not release stable until the spec becomes stable)?

Did not realize that https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#periodic-exporting-metricreader was still experimental! 

Will 1st check if Metrics Spec is close to marking this stable, and based on that device a plan for support in the SDK.

**Comment by pellared** [comment]
@cijothomas @reyang 
These env vars are to my best knowledge already supported by OTel Java, Python, and PHP. See [here](https://github.com/search?p=1&q=org%3Aopen-telemetry+OTEL_METRIC_EXPORT_INTERVAL&type=Code) and [here](https://github.com/search?p=1&q=org%3Aopen-telemetry+OTEL_METRIC_EXPORT_TIMEOUT&type=Code). I think the probability that this part of the spec would change is close to 0. 

I guess that `OTEL_METRICS_EXEMPLAR_FILTER` env var is the "more experimental" env var. This is the only reason why I am not creating a PR in the OTel spec 😉

**Review by Kielek — APPROVED** [review]

**Review by reyang — APPROVED** [review]
LGTM.

**Pull request merged**
